Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -149,10 +161,10 @@ CGColorSpaceRef CGColorGetColorSpace(CGColorRef color) {
 
 /**
  @Status Caveat
- @Notes Not all colors have 4 components, but all of the ones we currently support do!
+ @Notes Limited support for colorspace
 */
 size_t CGColorGetNumberOfComponents(CGColorRef color) {
-    return 4;
+    return CGColorSpaceGetNumberOfComponents(CGColorGetColorSpace(color)) + 1;
```

Comment:
this is based off of the limited support. Again a while fix to the colorspace is needed. 

---
In reply to: [106507058](https://github.com/Microsoft/WinObjC/pull/2244#discussion_r106507058) [](ancestors = 106507058)